src/physical/tabular/operations/triescan_minus.rs

```diff
@@ -1,14 +1,260 @@
+        debug_assert!(
+            tries_subtract
+                .iter()
+                .zip(infos.iter())
+                .all(|(trie_subtract, info)| info.used_layers.len()
+                    == trie_subtract.get_types().len())
+        );
+        debug_assert!(infos.iter().all(|info| info.used_layers.is_sorted()));
+        debug_assert!(infos.iter().all(|info| info
+            .used_layers
+            .iter()
+            .all(|&l| l < trie_main.get_types().len())));
+
+        let arity_main = trie_main.get_types().len();
+
+        let last_used_layers: Vec<usize> = infos
+            .iter()
+            .map(|info| {
+                *info
+                    .used_layers
+                    .iter()
+                    .max()
+                    .expect("There should be at least one layer that is used.")
+            })
+            .collect();
+        let mut column_scans = Vec::<UnsafeCell<ColumnScanT<'a>>>::with_capacity(arity_main);
+
+        for layer in 0..arity_main {
+            unsafe {
+                if let ColumnScanT::U64(left_scan_enum) = &*trie_main.get_scan(layer).unwrap().get()
+                {
+                    let mut scans_follower = Vec::<Option<&ColumnScanCell<'a, u64>>>::new();
+
+                    let mut subtract_indices = Vec::new();
+                    let mut follow_indices = Vec::new();
+
+                    for (subtract_index, (trie_subtract, info)) in
+                        tries_subtract.iter().zip(infos.iter()).enumerate()
+                    {
+                        let used_layer = info.used_layers.iter().position(|&l| l == layer);
+                        let is_last = layer == last_used_layers[subtract_index];
+
+                        if let Some(used_layer) = used_layer {
+                            if let ColumnScanT::U64(subtract_scan_enum) =
+                                &*trie_subtract.get_scan(used_layer).unwrap().get()
+                            {
+                                scans_follower.push(Some(subtract_scan_enum));
+                            }
+
+                            if is_last {
+                                subtract_indices.push(subtract_index);
+                            } else {
+                                follow_indices.push(subtract_index);
+                            }
+                        } else {
+                            scans_follower.push(None);
+                        }
+                    }
+
+                    let new_scan = ColumnScanEnum::ColumnScanSubtract(ColumnScanSubtract::new(
+                        left_scan_enum,
+                        scans_follower,
+                        subtract_indices,
+                        follow_indices,
+                    ));
+
+                    column_scans.push(UnsafeCell::new(ColumnScanT::U64(ColumnScanCell::new(
+                        new_scan,
+                    ))));
+                }
+            }
+        }
```

Comment:
Doesn't even require any additional `let`s, you can just push the `unsafe` inside: 
```rust
        for layer in 0..arity_main {
            if let ColumnScanT::U64(left_scan_enum) =
                &unsafe { *trie_main.get_scan(layer).unwrap().get() }
            {
                let mut scans_follower = Vec::<Option<&ColumnScanCell<'a, u64>>>::new();

                let mut subtract_indices = Vec::new();
                let mut follow_indices = Vec::new();

                for (subtract_index, (trie_subtract, info)) in
                    tries_subtract.iter().zip(infos.iter()).enumerate()
                {
                    let used_layer = info.used_layers.iter().position(|&l| l == layer);
                    let is_last = layer == last_used_layers[subtract_index];

                    if let Some(used_layer) = used_layer {
                        if let ColumnScanT::U64(subtract_scan_enum) =
                            &unsafe { *trie_subtract.get_scan(used_layer).unwrap().get() }
                        {
                            scans_follower.push(Some(subtract_scan_enum));
                        }

                        if is_last {
                            subtract_indices.push(subtract_index);
                        } else {
                            follow_indices.push(subtract_index);
                        }
                    } else {
                        scans_follower.push(None);
                    }
                }

                let new_scan = ColumnScanEnum::ColumnScanSubtract(ColumnScanSubtract::new(
                    left_scan_enum,
                    scans_follower,
                    subtract_indices,
                    follow_indices,
                ));

                column_scans.push(UnsafeCell::new(ColumnScanT::U64(ColumnScanCell::new(
                    new_scan,
                ))));
            }
        }
```